From 98ce0ae63a7330cc417b6b424f46ff487b8762ae Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Fri, 13 Nov 2020 13:50:45 +0800
Subject: [PATCH] drm/rockchip: vop2: Fix r2y/csc_mode reigster definition

Fix r2y_en and csc_mode_en bit definition of ESMART/SMART.

Change-Id: Ib41858d84161dfe20c4d2452506ce2ef1c22b6f4
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_vop2_reg.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c b/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c
index 587cf4e6f200..29881c58f839 100644
--- a/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c
+++ b/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c
@@ -731,8 +731,8 @@ static const struct vop2_win_regs rk3568_esmart_win_data = {
 	.yrgb_vir = VOP_REG(RK3568_ESMART0_REGION0_VIR, 0xffff, 0),
 	.uv_vir = VOP_REG(RK3568_ESMART0_REGION0_VIR, 0xffff, 16),
 	.y2r_en = VOP_REG(RK3568_ESMART0_CTRL0, 0x1, 0),
-	.r2y_en = VOP_REG(RK3568_ESMART0_CTRL0, 0x1, 9),
-	.csc_mode = VOP_REG(RK3568_ESMART0_CTRL0, 0x3, 10),
+	.r2y_en = VOP_REG(RK3568_ESMART0_CTRL0, 0x1, 1),
+	.csc_mode = VOP_REG(RK3568_ESMART0_CTRL0, 0x3, 2),
 };
 
 /*
-- 
2.35.3

